core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -268,218 +163,181 @@ static boolean hasOnlyPrintableAscii(String s) {
 
   @Nullable
   public String asId(int columnIndex, boolean required) {
-    String value = asString(columnIndex, required);
-    if (value != null && !hasOnlyPrintableAscii(value)) {
-      addNoticeInRow(
-          new NonAsciiOrNonPrintableCharNotice(
-              row.getFileName(), row.getRowNumber(), row.getColumnName(columnIndex)));
-    }
-    return value;
+    return asValidatedString(
+        columnIndex,
+        required,
+        RowParser::hasOnlyPrintableAscii,
+        NonAsciiOrNonPrintableCharNotice::new);
   }
 
   @Nullable
   public String asUrl(int columnIndex, boolean required) {
-    return urlParser.parseField(columnIndex, required);
+    return asValidatedString(
+        columnIndex, required, s -> UrlValidator.getInstance().isValid(s), InvalidUrlNotice::new);
   }
 
   @Nullable
   public String asEmail(int columnIndex, boolean required) {
-    return emailParser.parseField(columnIndex, required);
+    return asValidatedString(
+        columnIndex,
+        required,
+        s -> EmailValidator.getInstance().isValid(s),
+        InvalidEmailNotice::new);
   }
 
   @Nullable
   public String asPhoneNumber(int columnIndex, boolean required) {
-    return phoneNumberParser.parseField(columnIndex, required);
+    return asValidatedString(
+        columnIndex,
+        required,
+        s -> PhoneNumberUtil.getInstance().isPossibleNumber(s, feedName.getISOAlpha2CountryCode()),
+        InvalidPhoneNumberNotice::new);
   }
 
   @Nullable
   public Locale asLanguageCode(int columnIndex, boolean required) {
-    return languageCodeParser.parseField(columnIndex, required);
+    // FIXME: Enhance checks for IETF BCP 47 language code.
```

Comment:
@aababilov Could you open an issue on GitHub for this with a better description of how this should be enhanced?

Reply:
We don't actually need them, as far as I can tell. `Locale` fully supports IETF BCP 47. I added some unit tests to verify that.